crates/adapters/src/format/json/schema.rs

```diff
@@ -0,0 +1,429 @@
+            optional: schema.columntype.nullable,
+        })
+    }
+
+    fn type_schema(schema: &ColumnType) -> Result<Type, ControllerError> {
+        Ok(Type {
+            representation_type: representation_type_schema(schema)?,
+            logical_type: logical_type_schema(schema)?,
+        })
+    }
+
+    fn logical_type_schema(schema: &ColumnType) -> Result<Option<LogicalType>, ControllerError> {
+        Ok(match schema.typ.to_lowercase().as_str() {
+            "time" => Some(LogicalType {
+                name: "org.apache.kafka.connect.data.Time".to_string(),
+                parameters: BTreeMap::new(),
+            }),
+            "timestamp" => Some(LogicalType {
+                name: "org.apache.kafka.connect.data.Timestamp".to_string(),
+                parameters: BTreeMap::new(),
+            }),
+            "date" => Some(LogicalType {
+                name: "org.apache.kafka.connect.data.Date".to_string(),
+                parameters: BTreeMap::new(),
+            }),
+            _ => None,
+        })
+    }
+
+    fn representation_type_schema(
+        schema: &ColumnType,
+    ) -> Result<RepresentationType, ControllerError> {
+        match schema.typ.to_lowercase().as_str() {
+            "boolean" | "bool" => Ok(RepresentationType::Boolean),
+            "varchar" | "character varying" | "char" | "character" | "string" | "text" => {
+                Ok(RepresentationType::String)
+            }
+            "tinyint" => Ok(RepresentationType::Int8),
+            "smallint" | "int2" => Ok(RepresentationType::Int16),
+            "integer" | "int" | "signed" | "int4" => Ok(RepresentationType::Int32),
+            "bigint" | "int64" => Ok(RepresentationType::Int64),
+            // This requires Kafka connector to be configure with `"decimal.handling.mode":
+            // "string"`. Other encodings would require serde to know the scale of the
+            // decimal number, which isn't preserved in Rust.
+            "decimal" | "dec" | "numeric" | "number" => Ok(RepresentationType::String),
+            "real" | "float4" | "float32" => Ok(RepresentationType::Float),
+            "double" | "float8" | "float64" => Ok(RepresentationType::Double),
+            "time" => Ok(RepresentationType::Int64),
+            "timestamp" => Ok(RepresentationType::Int64),
+            "date" => Ok(RepresentationType::Int32),
+            "binary" | "varbinary" | "bytea" => Ok(RepresentationType::Bytes),
+            "array" => Ok(RepresentationType::Array {
+                items: Box::new(type_schema(schema.component.as_ref().ok_or_else(
+                    || {
+                        ControllerError::schema_validation_error(&format!(
+                            "element type is not specified for array type {schema:?}"
+                        ))
+                    },
+                )?)?),
+            }),
+            _ => Err(ControllerError::not_supported(&format!(
```

Comment:
intervals do not appear here.

Reply:
Yep, I didn't find a way to encode them in debezium. This requires special java classes to represent these logical types. Hmm, actually I just found it. Will fix.